File: articles/adolescence-identity-and-psychological-transition/go/adolescence_identity_summary.go
package main
import (
  "encoding/csv"; "fmt"; "log"; "os"; "strconv"
)
func idx(h []string, name string) int { for i,v := range h { if v == name { return i } }; return -1 }
func main() {
  if len(os.Args) < 2 { log.Fatal("usage: go run adolescence_identity_summary.go data/adolescence_identity_panel.csv") }
  f, err := os.Open(os.Args[1]); if err != nil { log.Fatal(err) }; defer f.Close()
  rows, err := csv.NewReader(f).ReadAll(); if err != nil { log.Fatal(err) }
  h := rows[0]; s:=idx(h,"identity_score"); c:=idx(h,"support_context"); x:=idx(h,"current_exclusion"); d:=idx(h,"digital_stress")
  n:=0; ss:=0.0; cc:=0.0; xx:=0.0; dd:=0.0
  for _, r := range rows[1:] { a,_:=strconv.ParseFloat(r[s],64); b,_:=strconv.ParseFloat(r[c],64); e,_:=strconv.ParseFloat(r[x],64); g,_:=strconv.ParseFloat(r[d],64); n++; ss+=a; cc+=b; xx+=e; dd+=g }
  fmt.Printf("Rows analyzed: %d\\nMean identity_score: %.4f\\nMean support_context: %.4f\\nMean current_exclusion: %.4f\\nMean digital_stress: %.4f\\n", n, ss/float64(n), cc/float64(n), xx/float64(n), dd/float64(n))
}

File: articles/adolescence-identity-and-psychological-transition/cpp/adolescence_identity_summary.cpp
#include <fstream>
#include <iostream>
#include <sstream>
#include <string>
#include <vector>
std::vector<std::string> split(const std::string& s){ std::vector<std::string> v; std::stringstream ss(s); std::string x; while(std::getline(ss,x,',')) v.push_back(x); return v; }
int idx(const std::vector<std::string>& h, const std::string& n){ for(size_t i=0;i<h.size();++i) if(h[i]==n) return (int)i; return -1; }
int main(int argc,char* argv[]){
  if(argc<2){ std::cerr<<"usage: adolescence_identity_summary data/adolescence_identity_panel.csv\\n"; return 1; }
  std::ifstream file(argv[1]); std::string line; std::getline(file,line); auto h=split(line);
  int s=idx(h,"identity_score"), c=idx(h,"support_context"), x=idx(h,"current_exclusion"), d=idx(h,"digital_stress");
  long n=0; double ss=0,cc=0,xx=0,dd=0;
  while(std::getline(file,line)){ auto v=split(line); ss+=std::stod(v[s]); cc+=std::stod(v[c]); xx+=std::stod(v[x]); dd+=std::stod(v[d]); n++; }
  std::cout<<"Rows analyzed: "<<n<<"\\nMean identity_score: "<<ss/n<<"\\nMean support_context: "<<cc/n<<"\\nMean current_exclusion: "<<xx/n<<"\\nMean digital_stress: "<<dd/n<<"\\n";
  return 0;
}
